File: C++/Leetcode/Array/removeElement.cpp
#include <iostream>
#include <vector>

class Solution{
    public:
        int removeElement(std::vector<int>& nums, int val){
            int l = 0;

            for (int i = 0; i < nums.size(); ++i){
                if (nums[i] != val){
                    nums[l] = nums[i];
                    l += 1;
                }
                else {
                    nums[l] = nums[i];
                }
            }

            return l;
        }
    
    protected:
        void printArray(std::vector<int>& nums){
            for (int i = 0; i < nums.size(); ++i){
                std::cout << nums[i] << std::endl;
            }
        }
};

int main(){
    Solution solution_instance;
    std::vector<int> nums = {3,2,2,3};
    solution_instance.removeElement(nums, 3);
    return 0;
};

File: Python/Leetcode/Array/removeElement.py
class Solution(object):
    def removeElement(self, nums, val):
        l = 0
        for r in range(len(nums)):
            if nums[r] != val:
                nums[l] = nums[r]
                l += 1
            else:
                nums[l] = nums[r]
                
        return l 

if __name__ == '__main__':
    solution_instance = Solution()
    solution_instance.removeElement([3,2,2,2,3], 3)